File: lab/latency/bench.py
import asyncio
from time import perf_counter

N = 1000000

async def worker():
    await asyncio.sleep(0.001)

async def main():
    start = perf_counter()
    await asyncio.gather(*[worker() for _ in range(N)])
    t = perf_counter() - start
    print(f"Python latency: {t}")

asyncio.run(main())

File: lab/latency/bench.go
package main

import (
	"fmt"
	"sync"
	"time"
)

const N = 1000000

func worker(wg *sync.WaitGroup) {
	time.Sleep(time.Millisecond)
	wg.Done()
}

func main() {
	var wg sync.WaitGroup

	start := time.Now()

	for i := 0; i < N; i++ {
		wg.Add(1)
		go worker(&wg)
	}

	wg.Wait()

	elapsed := time.Since(start)
	fmt.Printf("Go latency: %f\n", elapsed.Seconds())
}
